robot/board.py

```diff
@@ -41,6 +42,14 @@ def __init__(self, socket_path):
         self.data = b''
 
         self._connect(socket_path)
+        self._serial = Path(socket_path).stem
```

Comment:
Perhaps not relevant for this PR, but I would maybe wrap the `self.socket_path` as a `Path` which then may be of use further down the class. I notice in the `_connect` method we seem to be passing a `socket_path` but then the method just goes on to use the `self.socket_path` and then `str`s it, so it being a `Path` wouldn't be an issue.